Changed `server_url` so that it supports the full URL scheme

gsaslis opened the pull request:
WARNING: this is a breaking change!!

Also, I haven't modified the version / changelog, but I'd recommend moving to 1.1.0, to indicate that there is a breaking change and record this in the changelog too. Happy to add those commits as well if you want, it's just that some cookbook maintainers prefer to handle that themselves ;)

Commit:
Changed `server_url` so that it supports the full URL scheme

WARNING: this is a breaking change!!

Commit:
Changed `server_url` so that it supports the full URL scheme

WARNING: this is a breaking change!!

gsaslis commented:
Fixes #5

Commit:
Revert removal of port attribute, as it is needed elsewhere in config.js

Commit:
Revert removal of port attribute, as it is needed elsewhere in config.js

Commit:
Ensure install_dir exists

Commit:
Ensure install_dir exists

Commit:
Added `listen_address` attribute to overcome hard-coded `127.0.0.1` definition

Commit:
Added `listen_address` attribute to overcome hard-coded `127.0.0.1` definition

Commit:
Added `node_bin_path` attribute to make path to node executable configurable

This is necessary because in my setup node was installed (by the node cookbook) in the `/usr/local/bin/node` path, instead of the previously-hardcoded `/usr/bin/node`

Commit:
Added `node_bin_path` attribute to make path to node executable configurable

This is necessary because in my setup node was installed (by the node cookbook) in the `/usr/local/bin/node` path, instead of the previously-hardcoded `/usr/bin/node`

arukaen commented:
@gsaslis Thanks for this addition. Currently testing this before merging.

gsaslis commented:
Glad to hear! 
I had a requirement to install two ghost blogs on one server,  so I will be sending through a pull request shortly where I have converted the recipes into (chef 12.5) custom resources... just need to finish up the test kitchen work

arukaen commented on recipes/_ghost.rb:
```diff
@@ -4,6 +4,10 @@
     not_if { ::File.exist?("#{Chef::Config[:file_cache_path]}/ghost.zip") }
 end
 
+directory node['ghost-blog']['install_dir'] do
```
This breaks the conditional on [L#14](https://github.com/gsaslis/chef-ghost/blob/1e58310a0e7b949739aa84e7b0c82cc6de40ae46/recipes/_ghost.rb#L14). Any particular reason you want to create the directory before the `unzip` does?

gsaslis replied:
Yeah, my problem was that unless you included the nginx recipe, the parent dir didn't exist, so the unzip would fail. 

So, I added this check and (in a subsequent commit on my new branch, admittedly), changed the conditional to check for the `index.js` file. 
`not_if { ::File.exists?("#{install_dir}/index.js") }`

Does that sound ok to you?

arukaen replied:
Sounds fine with me. 👍

arukaen replied:
If you could just update the check in L#14 then I would be good with merging this in.

Commit:
node & kitchen updates

arukaen committed:
Update CHANGELOG.md

arukaen committed:
Update metadata.rb

arukaen committed:
Merge pull request #7 from arukaen/node-update

node & kitchen updates

gsaslis commented:
@arukaen when you get a chance, you might want to take a look at my custom_resource fork [here](https://github.com/gsaslis/chef-ghost/tree/feature/custom_resource) 

If it's along the same lines of what you had in mind for this cookbook, I'd be happy to open a PR

arukaen commented:
@gsaslis I checked your fork and it looks pretty awesome. If you want to contribute that functionality to the cookbook that would be awesome.

Commit:
Added fix for directory check

This broke after I explicitly added the `install_dir` directory creation to the cookbook, to protect against cases when the nginx recipe (that created the parent dir of the default `install_dir` did not exist and therefore the unzip would fail.

Commit:
Added fix for directory check

This broke after I explicitly added the `install_dir` directory creation to the cookbook, to protect against cases when the nginx recipe (that created the parent dir of the default `install_dir` did not exist and therefore the unzip would fail.

Commit:
Merge branch 'master' of https://github.com/gsaslis/chef-ghost

* 'master' of https://github.com/gsaslis/chef-ghost:
  Added fix for directory check
  Added `node_bin_path` attribute to make path to node executable configurable
  Added `listen_address` attribute to overcome hard-coded `127.0.0.1` definition
  Ensure install_dir exists
  Revert removal of port attribute, as it is needed elsewhere in config.js
  Changed `server_url` so that it supports the full URL scheme

gsaslis commented:
@arukaen I added the fix you requested. 

Let's merge that in, so I can then open a separate PR for the "v2.0.0" (?) rewrite?

arukaen commented:
LGTM 👍
